src/pip/_internal/legacy_resolve.py

```diff
@@ -141,8 +142,7 @@ def __init__(
         self.finder = finder
         self.session = session
 
-        # This is set in resolve
-        self.require_hashes = None  # type: Optional[bool]
+        self.require_hashes = require_hashes
```

Comment:
It looks like setting the value here above will cause the following assertion no longer to work as intended: https://github.com/pypa/pip/blob/828da22e02a012e1903298864b8c2cb570f82f48/src/pip/_internal/legacy_resolve.py#L288-L290

A couple ways around this might be:

1. Store `require_hashes_option` separate from the "effective" value.
2. Pass the option to `resolve()` instead of `__init__()`.

Reply:
As suggested, we now store `require_hashes_option` separate from `require_hashes` (see [here](https://github.com/pypa/pip/pull/7068/files#diff-a1f5543ad6553ad10d961f0234e3b51bR146)) and pass the "effective" value to the methods that need it (see [here](https://github.com/pypa/pip/pull/7068/commits/e41b5bc6cc7c5bb8596a2f17149f7fae20c52fc2)).